### ISSUE TITLE
Pervasives are deprecated - switch to stdlib-shims is suggested

### ISSUE THREAD

**Issue by XVilka** [BinaryAnalysisPlatform/bap#969]
https://github.com/BinaryAnalysisPlatform/bap/search?q=Pervasives&unscoped_q=Pervasives
```
- Alert deprecated: module Stdlib.Pervasives
- Use Stdlib instead.
- 
- If you need to stay compatible with OCaml < 4.07, you can use the 
- stdlib-shims library: https://github.com/ocaml/stdlib-shims
- File "setup.ml", line 3467, characters 16-34:
- 3467 |                 Pervasives.compare o2 o1)
-                        ^^^^^^^^^^^^^^^^^^
- Alert deprecated: module Stdlib.Pervasives
- Use Stdlib instead.
- 
- If you need to stay compatible with OCaml < 4.07, you can use the 
- stdlib-shims library: https://github.com/ocaml/stdlib-shims
[ERROR] The compilation of bap failed at
        "/home/travis/.opam/opam-init/hooks/sandbox.sh build ./configure
        --prefix=/home/travis/.opam/4.08.0 --with-cxx=`which clang++`
        --with-llvm-version=5.0.0git-929163d
        --with-llvm-config=/usr/local/clang-5.0.0/bin/llvm-config
        --mandir=/home/travis/.opam/4.08.0/man --enable-everything
        --enable-tests --disable-ida --disable-fsi-benchmark
        --disable-objdump".
#=== ERROR while compiling bap.master =========================================#
# context     2.0.4 | linux/x86_64 | ocaml-base-compiler.4.08.0 | pinned(file:///home/travis/build/BinaryAnalysisPlatform/bap)
# path        ~/.opam/4.08.0/.opam-switch/build/bap.master
# command     ~/.opam/opam-init/hooks/sandbox.sh build ./configure --prefix=/home/travis/.opam/4.08.0 --with-cxx=`which clang++` --with-llvm-version=5.0.0git-929163d --with-llvm-config=/usr/local/clang-5.0.0/bin/llvm-config --mandir=/home/travis/.opam/4.08.0/man --enable-everything --enable-tests --disable-ida --disable-fsi-benchmark --disable-objdump
# exit-code   1
# env-file    ~/.opam/log/bap-4270-05296d.env
# output-file ~/.opam/log/bap-4270-05296d.out
### output ###
# [...]
# Use Stdlib instead.
# 
# If you need to stay compatible with OCaml < 4.07, you can use the 
# stdlib-shims library: https://github.com/ocaml/stdlib-shims
# File "setup.ml", line 3467, characters 16-34:
# 3467 |                 Pervasives.compare o2 o1)
#                        ^^^^^^^^^^^^^^^^^^
# Alert deprecated: module Stdlib.Pervasives
# Use Stdlib instead.
# 
# If you need to stay compatible with OCaml < 4.07, you can use the 
# stdlib-shims library: https://github.com/ocaml/stdlib-shims
```
 from the 4.08 build log: https://travis-ci.org/BinaryAnalysisPlatform/bap/jobs/570684149

**Comment by ivg** [comment]
So, the tentative plan is the following, we will release 1.6.1 with bug fixes (therefore the minor version is not updated), and this would be the last pre 2.0 release (though, I imagine that in the future we will be releasing 1.6.x with critical bug fixes, but not on a regular basis).  Then we will merge yours #957 into the master, and stick to 4.07+ and core_kernel.v0.12 thereafter. That means, that we don't really need those shims and can update our code to use the `Stdlib` name instead of `Pervasives` (I believe this will make bap much faster, isn't it?)

I will let this idea to mature for some period of time, but otherwise #957 is good to go into master (after it is rebased on the current 2.0-ish master).

**Comment by ivg** [comment]
fixed in #957
